stablehlo/dialect/ChloOps.td

```diff
@@ -471,6 +471,21 @@ class CHLO_UnaryElementwiseOp<string mnemonic, list<Trait> traits,
   }];
 }
 
+def CHLO_AsinAcosKernelOp : CHLO_UnaryElementwiseOp<"asin_acos_kernel",
+    [HLO_CompatibleOperandsAndResultType], HLO_AnyComplexTensor> {
+  let summary = "AsinAcosKernel operator";
+
+  let description = [{
+    Returns `AsinAcosKernel(operand)` element-wise.
```

Comment:
Can we add more description:

"This op is used as an intermediate value in decompositions and should never be constructed directly by frameworks or consumed by backends."

And maybe rename this to `_asin_acos_kernel` if MLIR permits this.

Reply:
The alternative here would be to generate a CustomCall with the name `_asin_acos_kernel` and then write a pattern which decomposes that custom call. That may be more in-model, but as long as this op remains an impl detail, a refactor shouldn't be difficult if we want to go this route.

Reply:
I have updated asin_acos_kernel docs and renamed it to `_asin_acos_kernel`.

I like the custom call idea. IIUC, this should reduce chlo_legalize_to_stablehlo.mlir size as well.